### PR TITLE
feat(breaking change): experimental config options

### PR DESCRIPTION
**Pull request by diasdavid**
As recommended by @whyrusleeping, we need to pubsub behind a `EXPERIMENTAL` option in the config options.

Adding config options will cause a breaking cause in the API of creating a node, but it was a step we were going to make for #556 anyway.

### CODE REVIEW

File: src/core/components/index.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+exports.goOnline = require('./go-online')
```
Comment (dignifiedquire) [inline src/core/components/index.js]: let's not duplicate `exports` everywhere

```js
module.exports = {
  goOnline: require('./go-online'),
  ...
}
```

File: README.md [hunk 2]
```diff
@@ -200,7 +200,12 @@ The HTTP-API exposed by the js-ipfs daemon follows the [`http-api-spec`](https:/
 const repo = <IPFS Repo instance or repo path>
 
 // Create the IPFS node instance
-const node = new IPFS(repo)
+const node = new IPFS({
+  repo: repo,
+  EXPERIMENTAL: {
```
Comment (dignifiedquire) [inline README.md]: Do we have to upcase this? I think `experimental` is enough. What would be good is to print something to the console when this is enabled.

File: src/core/components/index.js (hunk 1 above)
Comment (diasdavid) [inline src/core/components/index.js]: Yeah, that way we have to worry about those commas, the way it is right now doesn't

File: README.md (hunk 2 above)
Comment (diasdavid) [inline README.md]: I like making things very OBVIOUS :D

Comment (dignifiedquire) [inline README.md]: BUT DO YOU REALLY NEED TO SCREAM AT ME?

File: src/core/components/index.js (hunk 1 above)
Comment (dignifiedquire) [inline src/core/components/index.js]: Not sure what commas you mean

Comment (diasdavid) [inline src/core/components/index.js]: if it is inside an object, for every add/delete, we get the 'no dangling comma situation'

File: README.md (hunk 2 above)
Comment (diasdavid) [inline README.md]: I'll add your log :)

Comment (diasdavid) [inline README.md]: added your log :)

File: src/core/components/index.js (hunk 1 above)
Comment (dignifiedquire) [inline src/core/components/index.js]: we have a lot of mixed and unclean diffs, not sure this makes any **diff** ;)

**Pull request merged**